src/LLVM_Runtime_Linker.cpp

```diff
@@ -726,7 +732,11 @@ std::unique_ptr<llvm::Module> get_initial_module_for_target(Target t, llvm::LLVM
             modules.push_back(get_initmod_gpu_device_selection(c, bits_64, debug));
             modules.push_back(get_initmod_tracing(c, bits_64, debug));
             modules.push_back(get_initmod_write_debug_image(c, bits_64, debug));
-            modules.push_back(get_initmod_cache(c, bits_64, debug));
+            if (t.arch != Target::Hexagon) {
+                // TODO: Support this module in the Hexagon backend,
```

Comment:
I don't think so. This never would have worked before, it just used to get stripped from the runtime, now it doesn't, so we hit the issue.